README.rst

```diff
@@ -289,6 +289,18 @@ Choose from the list of available rules:
   Ensure there is no code on the same line as the PHP open tag and it is
   followed by a blank line.
 
+* **blank_line_around_class_body**
+
+  Ensure that class body contains one blank line after class definition
+  and before its end.
+
+  Configuration options:
+
+  - ``apply_to_anonymous_classes`` (``bool``): whether this fixer should be applied
+    to anonymous classes; defaults to ``false``
```

Comment:
```php
$classInstance = new class () {


   public function foo() { }



}
```

why allowing for not applying the change here ?

Reply:
Oh, I missed this. It should be `true` by default but must be configurable to support our code style (in anonymous classes we do not leave empty lines).

Reply:
sounds messy. why different rule depends is it anonymous or not?
what's the case? could you show examples? I can imagine one-liner anon class that we don't want to break to multiple lines, but, especially with example I gave, why not fix it ?

Reply:
> why different rule depends is it anonymous or not?

Because it's common to us to have code like this:

```php
<?php
class TestAmqpComponent extends \common\components\RabbitMQ\Component {

    public function getConnection() {
        return new class extends \PhpAmqpLib\Connection\AbstractConnection {
            public function __construct(
                $user,
                $password,
                $vhost,
                $insist,
                $login_method,
                $login_response,
                $locale,
                \PhpAmqpLib\Wire\IO\AbstractIO $io,
                $heartbeat
            ) {
                // do nothing
            }
        };
    }

}
```

Or like this (sorry for a screenshot):

![](http://erickskrauch.ely.by/monosnap-10-04-2018-at-21-20-33.png)

We don't make whitespace for anonymous classes 'cause it's mainly inlined in some another code and feels like a simple statement and not really class definition, so it should be inlined.

-------------

But as I said earlier this fixer should be applied to anonymous classes by default. Maybe better I should replace it with `anonymous_class_blank_lines_count`?

Reply:
sounds better, to not have hack to opt the feature out for some cases, but simply has config for anon and config for non-anon classes.